src/tox/config/loader/ini/factor.py

```diff
@@ -19,11 +19,7 @@ def filter_for_env(value: str, name: Optional[str]) -> str:
                 overall.append(content)
         else:
             for group in factors:
-                for a_name, negate in group:
-                    contains = a_name in current
-                    if not ((contains is True and negate is False) or (contains is False and negate is True)):
-                        break
-                else:
+                if all((a_name in current) ^ negate for a_name, negate in group):
```

Comment:
This is one of those now it's shorter but harder to read. I mean I'm not against keeping this, but just looking at it I'm not 100% it's correct, though I trust the test suite.

Reply:
Please decide what fits you. From my point of view, my proposal is slightly slower to read per word but more compact and therefore needs less time and mental energy to understand. But that's my point, so please feel free to decide.